test/functional/organizations.js

```diff
@@ -0,0 +1,90 @@
+'use strict';
+
+require('../testHelpers');
+
+exports.lab = Lab.script();
+const server = Server.server;
+
+let Organization;
+
+describe('Organizations resources', () => {
+  before(async () => {
+    await Server.initialize();
+    ({Organization} = server.models);
+  });
+
+  describe('Initial values', () => {
+    let organizations;
+    before(async () => {
+      organizations = await Organization.listOrganizations();
+    });
+
+    it('should have a Organization name 1', (done) => {
+      expect(organizations[0].name).to.equal('Organization name 1');
+      return done();
+    });
+
+    it('should have a Organization name 2', (done) => {
+      expect(organizations[1].name).to.equal('Organization name 2');
+      return done();
+    });
+
+    it('should have a Organization name 3', (done) => {
+      expect(organizations[2].name).to.equal('Organization name 3');
+      return done();
+    });
+  });
+
+  describe('GET /organizations', () => {
+    function callServer(test) {
+      return runTest({
+        method: 'GET',
+        url: '/organizations',
+        headers: {
+          'Device': 'client'
+        }
+      }, test);
+    }
+
+    it('should return a list of organizations', () => {
+      return callServer(({statusCode, result}) => {
+        expect(statusCode).to.equal(200);
+        expect(result).to.equal([{
```

Comment:
If you need to write organizations for test, create them on the setup. You can have a beforeAll where you set them, or you can use the createTestSetup helper that will create a couple of organizations and users: https://github.com/adoptapanama/admin-api/blob/develop/routes/users.js#L29-L39

You can see an example of how I'm using it here: https://github.com/adoptapanama/admin-api/blob/develop/test/functional/users.js#L15-L21

Just make sure you do a cleanup afterwards like this: https://github.com/adoptapanama/admin-api/blob/develop/test/functional/users.js#L23-L25

Reply:
ok, i will edit it, but what is the different between make the test in another file and write user, role and organization in createTestSetup() in users.js

Reply:
Most of the routes are going to be working behind the policy plugin, meaning only certain users should be able to access that resource. However, there's a lot of boilerplate that needs to be done for a user to be able to test his permissions. For example, a user belongs to several organizations and it can have different roles in different organizations, and each role has a specific set of permissions, but the user can only be logged to one role at at time. This means that for you to test, you need to create a sample organization, a sample role, a set of permissions and attach them to the role, a user, and then attach the user to the role and the organization.

It is a lot of work to be done in every test file. Thats why we have a helper called createTestSetup that does all that boilerplate, so that you can use the users, roles, and organizations created by the boilerplate in your test without having to write a lot of code, that may even change if new features are introduced.

But just as you do with any setup, you need to delete everything created by the boilerplate (the cleanup) with `resetTestSetup` to ensure that all those models created won't affect other tests in the app.

Reply:
oh i understand, ty